### PR TITLE
Make async_scan scan each opened directory asynchronously

### PR DESCRIPTION
**Pull request by mauriciopoppe**
This is an implementation of the comment in https://github.com/nvim-neo-tree/neo-tree.nvim/issues/236#issuecomment-1088210652, pasting the algorithm for future reference:

```
local function async_scan(context, callback /* new */)
  for every path in context.paths_to_load
    async_scan(path) ->
        scan.scan_dir_async(parent) -> 
        wait for on_exit callback ->  increase counter and call `callback` when all the paths in context_paths are done
```

### CODE REVIEW

**Comment by cseickel** [comment]
Oh cool, you completely dropped plenary's scan_dir from this. I'll give this a try and merge it if there are no problems. I do have a lot of other things up in the air though so this might sit a bit, now that we know this is not the primary slowdown.

File: lua/neo-tree/sources/filesystem/lib/fs_scan.lua [hunk 1]
```diff
@@ -9,13 +10,17 @@ local log = require("neo-tree.log")
...
   local state = context.state
-  local paths_to_load = context.paths_to_load
-  local folders = context.folders
-
+  local scanned_folder = context.folders[current_dir]
+  if scanned_folder then
+    scanned_folder.loaded = true
+  end
   if state.use_libuv_file_watcher then
     local root = context.folders[dir_path]
```
Comment (cseickel) [inline lua/neo-tree/sources/filesystem/lib/fs_scan.lua:25]: I think `dir_path` needs to be updated to `current_dir`. You'll need to test with the `use_libuv_file_watcher = true` to make sure it is working.

Comment (mauriciopoppe) [reply lua/neo-tree/sources/filesystem/lib/fs_scan.lua:25]: good catch, thanks

**Review by cseickel — APPROVED** [review]

**Comment by cseickel** [comment]
Looks good, thanks!

**Pull request merged**